src/components/DevToolsUi.tsx

```diff
@@ -5,80 +5,125 @@ import { Tabs, TabsContent, TabsList, TabsTrigger } from '@/components/ui/tabs'
               <TabsTrigger value="raw">Raw Data</TabsTrigger>
             </TabsList>
           </div>
 
-          <TabsContent value="basicInfo" className="h-[calc(100%-3rem)]">
-            <MetadataTable metadata={diagnosticData.basicInfo} />
-          </TabsContent>
-          <TabsContent value="entityInfo" className="h-[calc(100%-3rem)]">
-            <MetadataTable metadata={diagnosticData.entityInfo} />
-          </TabsContent>
+          {diagnosticData.bugReportDetails && (
+            <TabsContent value="details" className="h-[calc(100%-3rem)]">
+              <RawContent content={diagnosticData.bugReportDetails} />
+            </TabsContent>
+          )}
+          {diagnosticData.backendErrors && diagnosticData.backendErrors.length > 0 && (
+            <TabsContent value="backendErrors" className="h-[calc(100%-3rem)]">
+              <LogsTable logs={diagnosticData.backendErrors} title="Backend Errors" />
+            </TabsContent>
+          )}
+          {diagnosticData.userLogs && diagnosticData.userLogs.length > 0 && (
+            <TabsContent value="userLogs" className="h-[calc(100%-3rem)]">
+              <LogsTable logs={diagnosticData.userLogs} title="User Logs" />
+            </TabsContent>
+          )}
+          {diagnosticData.logs && diagnosticData.logs.length > 0 && (
+            <TabsContent value="logs" className="h-[calc(100%-3rem)]">
+              <LogsTable logs={diagnosticData.logs} title="System Logs" />
+            </TabsContent>
+          )}
           <TabsContent value="frontendErrors" className="h-[calc(100%-3rem)]">
             <ConsoleOutput
               errors={diagnosticData.frontendErrors}
               onErrorCountChange={setFrontendErrorCount}
             />
           </TabsContent>
-          <TabsContent value="backendErrors" className="h-[calc(100%-3rem)]">
-            <LogsTable logs={diagnosticData.backendErrors} title="Backend Errors" />
-          </TabsContent>
-          <TabsContent value="userLogs" className="h-[calc(100%-3rem)]">
-            <LogsTable logs={diagnosticData.userLogs} title="User Logs" />
-          </TabsContent>
-          <TabsContent value="logs" className="h-[calc(100%-3rem)]">
-            <LogsTable logs={diagnosticData.logs} title="System Logs" />
-          </TabsContent>
-          <TabsContent value="queryResults" className="h-[calc(100%-3rem)]">
-            <QueryResults data={diagnosticData.queryResults} />
-          </TabsContent>
+          {diagnosticData.entityInfo && (
+            <TabsContent value="entity" className="h-[calc(100%-3rem)]">
+              <RawContent content={diagnosticData.entityInfo} />
+            </TabsContent>
+          )}
+          {diagnosticData.queryResults && diagnosticData.queryResults.row_count > 0 && (
+            <TabsContent value="queryResults" className="h-[calc(100%-3rem)]">
+              <QueryResultsPanel data={diagnosticData.queryResults} />
+            </TabsContent>
+          )}
```

Comment:
I don't think we want to hide this for queries without any results. We could get really useful information out of this data even when zero rows return.

Reply:
Yep, valid point! Reverted